src/AppBundle/Entity/RepositoryStarListener.php

```diff
@@ -26,13 +35,18 @@ public function decStarsCount(RepositoryStar $repositoryStar, LifecycleEventArgs
 
     protected function updateStarsCount(RepositoryStar $repositoryStar, EntityManager $em, $operator)
     {
+        $repository = $repositoryStar->getRepository();
+
         $qb = $em->createQueryBuilder()
             ->update(Repository::class, 'r')
             ->set('r.stars', "r.stars $operator 1")
             ->where('r.id = :id')
-            ->setParameter('id', $repositoryStar->getRepository()->getId())
+            ->setParameter('id', $repository->getId())
             ->getQuery();
+        $qb->execute();
+
+        $em->refresh($repository);
 
-        return $qb->execute();
+        $this->elasticaObjectPersister->replaceOne($repository);
```

Comment:
Only RepositoryStar entity is updated here. But Elastica only listen to Repository entity changes.

I tried to mark entity as "scheduledUpdate" in UOW but Doctrine refuse to update a unchanged entity. So I had to update Repository manually.